src/common/actions/repository-input.js

```diff
@@ -0,0 +1,69 @@
+import 'isomorphic-fetch';
+import parseGitHubUrl from 'parse-github-url';
+
+export const CHANGE_REPOSITORY_INPUT = 'CHANGE_REPOSITORY_INPUT';
+export const UPDATE_STATUS_MESSAGE = 'UPDATE_STATUS_MESSAGE';
+export const VERIFY_GITHUB_REPOSITORY = 'VERIFY_GITHUB_REPOSITORY';
+export const VERIFY_GITHUB_REPOSITORY_SUCCESS = 'VERIFY_GITHUB_REPOSITORY_SUCCESS';
+export const VERIFY_GITHUB_REPOSITORY_ERROR = 'VERIFY_GITHUB_REPOSITORY_ERROR';
+
+export function changeRepositoryInput(value) {
+  return {
+    type: CHANGE_REPOSITORY_INPUT,
+    payload: value
+  };
+}
+
+export function updateStatusMessage(message) {
+  return {
+    type: UPDATE_STATUS_MESSAGE,
+    payload: message
+  };
+}
+
+function checkStatus(response) {
+  if (response.status >= 200 && response.status < 300) {
+    return response;
+  } else {
+    const error = new Error(response.statusText);
+    error.response = response;
+    throw error;
+  }
+}
+
+export function verifyGitHubRepository(repository) {
+  return (dispatch) => {
+    const gitHubRepo = parseGitHubUrl(repository);
+    const repo = gitHubRepo ? gitHubRepo.repo : null;
+
+    if (repo) {
+      dispatch({
+        type: VERIFY_GITHUB_REPOSITORY,
+        payload: gitHubRepo.repo
+      });
+
+      return fetch(`https://api.github.com/repos/${repo}/contents/snapcraft.yaml`)
+        .then(checkStatus)
+        .then(() => dispatch(verifyGitHubRepositorySuccess(`https://github.com/${repo}.git`)))
+        .catch(error => dispatch(verifyGitHubRepositoryError(error)));
+    } else {
+      // TODO: above we return promise, here nothing - inconsistent and harder to test
```

Comment:
Not sure how to best handle this.

The problem is that `verifyGitHubRepository` does 2 steps of verification.
One is synchronous (parse the input with `parseGitHubUrl` and check if it's valid GH repo URL) and the other is asynchronous (call GH API for snapcraft.yaml).

In async branch we return the fetch promise (so it can be tested). The sync branch just dispatches failure action immediately. This means both branches need different types of tests and the return value of this function is inconsistent.

I'm not sure what would be the best way to solve it? Wrapping sync branch in immediately resolved Promise works, but feels unnecessary and bit dirty. Maybe splitting that into 2 separate actions (one sync for just parsing the input and the other for async calling API)? Any other ideas?